include/wx/generic/dirctrlg.h

```diff
@@ -194,18 +219,32 @@ class WXDLLIMPEXP_CORE wxGenericDirCtrl: public wxControl
     bool ExtractWildcard(const wxString& filterStr, int n, wxString& filter, wxString& description);
 
 private:
+    int  GetAvailableID() { return m_availableID++; }
     void PopulateNode(wxTreeItemId node);
     wxDirItemData* GetItemData(wxTreeItemId itemId);
-
-    bool            m_showHidden;
-    wxTreeItemId    m_rootId;
-    wxString        m_defaultPath; // Starting path
-    long            m_styleEx; // Extended style
-    wxString        m_filter;  // Wildcards in same format as per wxFileDialog
-    int             m_currentFilter; // The current filter index
-    wxString        m_currentFilterStr; // Current filter string
-    wxTreeCtrl*     m_treeCtrl;
-    wxDirFilterListCtrl* m_filterListCtrl;
+    void AddRightClickMenuItem(const wxString& label, void(wxGenericDirCtrl::*function)(wxCommandEvent &));
+    void MenuRename(wxCommandEvent & evt);
+    void MenuSortAlpha(wxCommandEvent & evt);
+    void MenuSortNameReversed(wxCommandEvent & evt);
+    void MenuSortNatural(wxCommandEvent & evt);
+    void MenuSortDate(wxCommandEvent & evt);
+    void MenuSortDateReverse(wxCommandEvent & evt);
+    void HandleDirMenu();
+    void HandleFileMenu();
+
+    bool                    m_showHidden;
+    wxTreeItemId            m_rootId;
+    wxTreeItemId            m_rightClickedItemId;
```

Comment:
This should be `m_itemPopupMenu` or something similar (i.e. again use "popup" instead of "right click").